observer.go

```diff
@@ -0,0 +1,15 @@
+package opentracing
+
+// Observer can be registered with the Tracer to recieve notifications
```

Comment:
Hi @bhs, I have added a commit explaining how its registered, in the comments, for the observer api.
Thanks for the review!